src/main/scala/io/citrine/lolo/Model.scala

```diff
@@ -9,7 +11,21 @@ trait Model[+T <: PredictionResult[Any]] extends Serializable {
     * Apply the model to a seq of inputs
     *
     * @param inputs to apply the model to
-    * @return a predictionresult which includes, at least, the expected outputs
+    * @return a PredictionResult which includes, at least, the expected outputs
     */
   def transform(inputs: Seq[Vector[Any]]): T
+
+  /**
+    * Compute Shapley feature attributions for a given input in this node's subtree
+    *
+    * @param input for which to compute feature attributions.
+    * @param omitFeatures feature indices to omit in computing Shapley values
+    * @return array of vector-valued attributions for each feature
+    *         One DenseVector[Double] per feature, each of length equal to the output dimension.
+    */
+  def shapley(
+               input: Vector[Any],
+               omitFeatures: Set[Int] = Set()
+             ): Option[Vector[DenseVector[Double]]] = None
```

Comment:
Why specify the internal vector as dense?

Reply:
This is specified as a breeze vector so that it can be directly used with linalg operations, especially vector addition.